docker/Dockerfile

```diff
@@ -1,23 +1,24 @@
-FROM ubuntu:xenial
+FROM alpine:3.6
 
-RUN apt-get update -y && apt-get upgrade -y && apt-get install -y \
-build-essential \
+RUN apk upgrade && apk update && apk add \
+g++ \
+make \
 cmake \
-libboost-dev \
-libboost-filesystem-dev \
-libboost-program-options-dev \
-libboost-regex-dev \
-libsqlite3-dev \
+boost-dev \
+boost-filesystem \
+boost-program_options \
+boost-regex \
+sqlite-dev \
 ragel \
 wget
 
-# Ubuntu-native ODB packages don't seem to work due to a broken ABI, see this thread: http://codesynthesis.com/pipermail/odb-users/2016-May/003277.html
-RUN wget http://www.codesynthesis.com/download/odb/2.4/odb_2.4.0-1_amd64.deb -O /opt/odb_2.4.0-1_amd64.deb
+RUN mkdir /opt
+RUN wget http://www.codesynthesis.com/download/odb/2.4/odb-2.4.0.tar.bz2 -O /opt/odb.tar.bz2
 RUN wget http://codesynthesis.com/download/odb/2.4/libodb-2.4.0.tar.bz2 -O /opt/libodb.tar.bz2
 RUN wget http://codesynthesis.com/download/odb/2.4/libodb-sqlite-2.4.0.tar.bz2 -O /opt/libodb-sqlite.tar.bz2
 
 # ODB compiler
-RUN cd /opt && tar jxvf libodb.tar.bz2 && tar jxvf libodb-sqlite.tar.bz2 && dpkg -i odb_2.4.0-1_amd64.deb
+RUN cd /opt && tar jxvf odb.tar.bz2 && tar jxvf libodb.tar.bz2 && tar jxvf libodb-sqlite.tar.bz2
 RUN cd /opt/libodb-2.4.0 && ./configure && make && make install
 RUN cd /opt/libodb-sqlite-2.4.0 && ./configure --with-libodb=../libodb-2.4.0 && make && make install
```

Comment:
I wonder if removing the downloaded packages at the end of the Dockerfile would shrink the image even more... 🤔

Reply:
I'll try it :)

Reply:
Removing wget and the ODB downloads reduced the size by only 2 MBs (from 516mb to 514mb)

Reply:
I think that as the installation of those programs already copies the required files to the respective system directory those /opt directories could be safely deleted altogether. With that you should reduce around 50mb of the resulting size.

But if you really want to reduce layer sizes then the download, compilation and delete should be on the same run command together, ugly as hell, but it's the sad state of the art.

Another thing that could be checked is that after the apk operations you should run a apk clean to remove the package caches, not sure if alpine preserves copies of updated packages (that is another typical thing) and finally if you still have a man directory you can safely delete it.

Reply:
> But if you really want to reduce layer sizes then the download, compilation and delete should be on the same run command together, ugly as hell, but it's the sad state of the art.

Do we need to get to that extreme? It is recommended to minimize the number of layers, but we are really really far from the limit. We have already reduced the size quite a lot so I would prefer not to make the Dockerfile horrible to read.

Reply:
The problem it's that to download a image you have to download all the layers, if you do a layer when you create a lot of information and another layer when you remove all that information... It's like yes, your final image weights X but the effective size of all the layer to download is X+Y

That is why the pattern download / configure / compile delete in the same run statement the layer will only be the final diff between the previous layer and the current state of the machine. This doesn't mean that you need to put everything together just the stuff that is related. The alternative are the intermediate builder dockers that are starting to appear but it will probably be too much hassle as they are kind of beta now.